tests/errmsgs/t18886.nim

```diff
@@ -0,0 +1,18 @@
+discard """
+  cmd: "nim check --hints:off $file"
+  errormsg: ""
+  nimout: '''
+t18886.nim(18, 24) Error: ambiguous identifier: 'bar' -- use one of the following:
+  t18886.bar: proc (i: ptr int){.noSideEffect, gcsafe, locks: 0.}
+  t18886.bar: proc (i: ptr char){.noSideEffect, gcsafe, locks: 0.}
+'''
```

Comment:
But you cannot use this syntax instead at all, this error message is highly misleading!

Reply:
so what's the expected behavior, what I thought is `cast` get ambiguous identifier and the error message point out where they are.

Reply:
implemente a new feature support choose between two `bar` ?

Reply:
Make the error message better. Like "ambiguous identifier, you need a helper proc to disambiguate"